packages/cdktf-cli/lib/models/terraform-cloud.ts

```diff
@@ -306,8 +310,17 @@ export class TerraformCloud implements Terraform {
       // In rare cases the backend sends an empty chunk of data back.
       if (data && data.length) {
         const buffer = Buffer.from(data, "utf8");
+
         stdout(buffer);
-        sendLog(buffer.toString("utf8"), false);
+
+        // We only want to send what we have not seen yet.
+        const bufferWithoutLastKnown = buffer
+          .toString("utf8")
+          .replace(this.lastLogMessage, "");
+        if (bufferWithoutLastKnown.trim().length) {
+          sendLog(bufferWithoutLastKnown, false);
+          this.lastLogMessage = buffer.toString("utf8");
+        }
```

Comment:
Couldn't we also use`Buffer.subarray` and store the previous length instead?
Or is there any reason why to use string manipulation for this?

Reply:
This might feel like a premature optimization, but to be honest the string replacing also feels a bit hacky to me.